manifests/config.pp

```diff
@@ -27,9 +27,15 @@
 #     ensure => 'blank',
 #   }
 #
+#   # By value type/size:
+#   postfix::config {
+#     'myorigin':      value => undef  # Rm
+#     'myhostname':    value => 'xyz'  # Set 
+#     'mydestination': value => '',    # Clear
+#   }
 define postfix::config (
-  Optional[String]                   $value  = undef,
-  Enum['present', 'absent', 'blank'] $ensure = 'present',
+  Variant[Undef, String] $value  = undef,
```

Comment:
What is the need to change from `Optional` to `Variant`?

Reply:
I believe none. It was a first thought, and tested, however should work the same.

Reply:
Yes, it's equivalent, but `Optional` is the recommended way of doing this, so I'd rather stick to it.

Reply:
Sure. I'll change. Optional for Enum then as well. For same style
Optional[Enum['present', 'absent', 'blank']] $ensure = undef,

Reply:
After reading your code again, I'm not sure making `ensure` optional is the best approach. What you're adding essentially is support for empty values, managed by `clear`ing the value in the Augeas tree.

So instead, I'd suggest to simply have a case for `$value == ""` in the `ensure == 'present'` case below, wherein you'll use `clear` instead of `set`. This should be the only required change for your need, since `ensure` defaults to `present` anyway.

Reply:
Not only empty. If `value == undef` it would act as `'rm'` command or `ensure == absent` option, thus it would conflict with default `ensure == 'present'`. The idea is to omit this option in case if you are not sure what would be passed by hiera in future. Like in my example in first [comment](https://github.com/camptocamp/puppet-postfix/pull/290#issue-541196745)